### PR TITLE
isort convolution

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/convolution/setup_package.py [hunk 1]
```diff
@@ -3,9 +3,8 @@
 import os
 import sys
 
-from setuptools import Extension
```
Comment (pllim) [inline astropy/convolution/setup_package.py:6]: Doesn't it make more sense to import setuptools before you import numpy for this case?

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: IDK, why?

Comment (pllim) [reply astropy/convolution/setup_package.py:6]: setuptools is from Python org and numpy is our own third-party dep. I think the existing order makes more sense.

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: Oh. I don't think isort distinguishes between third-party dependencies. We could add a category between `STDLIB` and `THIRD-PARTY`, like `INFRASTRUCTURE`...

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: Though I'd like to get more discussion before making that more drastic change. @mhvk @WilliamJamieson @eerovaher @saimn (just some people recently in these formatting discussions).

Comment (mhvk) [reply astropy/convolution/setup_package.py:6]: I realize I thought `setuptools` was in the standard library, which it is not...  My sense would be to just let `isort` do its work even though @pllim's comment feels right, that it should be earlier.

**Comment by nstarman** [comment]
Rebased to fix merge conflict.

File: astropy/convolution/setup_package.py (hunk 1 above)
Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: We have a ``build_requires`` in the ``pyproject.toml``, which feels like a subcategory of third-party dependencies, so  I'm either way on separating these out.

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: Vote tally
- keep.   :  +1 @mhvk, +.5 @nstarman 
- change: +1 @pllim,  +.5 @nstarman

Comment (WilliamJamieson) [reply astropy/convolution/setup_package.py:6]: My vote is for the simplest isort configuration possible.

Comment (mhvk) [reply astropy/convolution/setup_package.py:6]: Really more like +0.3 or so... But simplest configuration possible is attractive.

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: @pllim how about we leave isort as is for now and open an issue to discuss future refinement? With @WilliamJamieson's PR to auto run pre-commit any decisions will propagate to the whole repo 🥳

Comment (nstarman) [reply astropy/convolution/setup_package.py:6]: #13718 if we want to discuss before/during/after this PR is merged.

Comment (pllim) [reply astropy/convolution/setup_package.py:6]: Sure, CI passed after all, so not like the order really matters; just a human preference.

**Pull request merged**